recipes/occt/all/conanfile.py

```diff
@@ -0,0 +1,112 @@
+        extracted_dir = self.name.upper() + "-" + \
+            self.version.replace(".", "_")
+        tools.rename(extracted_dir, self._source_subfolder)
+
+    def _patch_sources(self):
+        tools.replace_in_file(
+            os.path.join(self._source_subfolder, "CMakeLists.txt"),
+            "project (OCCT)",
+            '''project (OCCT)
+                include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
+                conan_basic_setup()''')
+        tcl_libs = self.deps_cpp_info["tcl"].libs
+        tcl_lib = next(filter(lambda lib: "tcl8" in lib, tcl_libs))
+        tools.replace_in_file(
+            os.path.join(self._source_subfolder, "adm/cmake/tcl.cmake"),
+            "${CSF_TclLibs}",
+            tcl_lib)
+
+        tk_libs = self.deps_cpp_info["tk"].libs
+        tk_lib = next(filter(lambda lib: "tk8" in lib, tk_libs))
+        tools.replace_in_file(
+            os.path.join(self._source_subfolder, "adm/cmake/tk.cmake"),
+            "${CSF_TclTkLibs}",
+            tk_lib)
+
+    def _configure_cmake(self):
+        if self._cmake:
+            return self._cmake
+        self._cmake = CMake(self)
+        self._cmake.definitions["3RDPARTY_TCL_LIBRARY_DIR"] = \
+            os.path.join(self.deps_cpp_info["tcl"].rootpath, "lib")
+        self._cmake.definitions["3RDPARTY_TCL_INCLUDE_DIR"] = \
+            self.deps_cpp_info["tcl"].include_paths[0]
+        self._cmake.definitions["3RDPARTY_TK_LIBRARY_DIR"] = \
+            os.path.join(self.deps_cpp_info["tk"].rootpath, "lib")
+        self._cmake.definitions["3RDPARTY_TK_INCLUDE_DIR"] = \
+            self.deps_cpp_info["tk"].include_paths[0]
+        if not self.options.shared:
+            self._cmake.definitions["BUILD_LIBRARY_TYPE"] = "Static"
+
+        self._cmake.definitions["INSTALL_DIR_BIN"] = "bin"
+        self._cmake.definitions["INSTALL_DIR_INCLUDE"] = "include"
+        self._cmake.definitions["INSTALL_DIR_LIB"] = "lib"
+        self._cmake.definitions["INSTALL_DIR_RESOURCE"] = "res"
+
+        self._cmake.configure(source_folder=self._source_subfolder)
+        return self._cmake
+
+    def build(self):
+        if self.options.shared:
+            if not self.options["tcl"].shared or \
+                    not self.options["tk"].shared or \
+                    not self.options["freetype"].shared:
+                raise ConanInvalidConfiguration(
+                    "tcl, tk and freetype must be shared when occt is shared.")
+        self._patch_sources()
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        cmake = self._configure_cmake()
+        cmake.install()
+        self.copy(
+            "LICENSE_LGPL_21.txt",
+            src=self._source_subfolder,
+            dst="licenses")
+        self.copy(
+            "OCCT_LGPL_EXCEPTION.txt",
+            src=self._source_subfolder,
+            dst="licenses")
+        with tools.chdir(self.package_folder):
```

Comment:
```suggestion
        with tools.chdir(self.package_folder):
```

Be explicit with remove, so we don't need to guess what this recipe does. Take a look on other recipes on CCI. Use [tools.rmdir](https://docs.conan.io/en/latest/reference/tools.html#tools-mkdir-tools-rmdir) instead.